app/components/records-charter.js

```diff
@@ -120,8 +244,40 @@ export default Component.extend({
     return dataset;
   },
 
-  randomUpto(size) {
-    return Math.floor(Math.random() * size);
+  zip(arrayOfArrays, delimiter = '/') {
+    let zipped = arrayOfArrays[0].map((_, i) => arrayOfArrays.map(a => a[i]));
+    return zipped.map(a => this.join(a, delimiter));
+  },
+
+  join(array, delimiter) {
+    return array.reduce((p, c) => `${p}${delimiter}${c}`);
+  },
+
+  getColumnValues(column, rows) {
+    return rows.map(row => row[column]);
+  },
+
+  getJoinedRowValues(columns, row, delimiter = '/') {
+    return this.join(columns.map(column => row[column]), delimiter);
+  },
+
+  fixedColors(names) {
+    let colors = [];
+    for (let i = 0; i < names.length; ++i) {
+      colors.push(this.fixedColor(names[i]));
+    }
+    return colors;
+  },
+
+  fixedColor(atom) {
+    // Multiply by a large prime if number. Then stick the result into a string regardless.
+    let string = `${isEqual(typeOf(atom), 'number') ? atom * 104729 : atom}`;
+    let hash = this.hash(string);
+    let colors = [];
+    for (let i = 0; i <= 2; ++i, hash = hash >> 8) {
+      colors.push(hash >> 8 & 0xFF)
```

Comment:
There are many lines in the graph with the same color. Maybe you should not (>> 8) here?